satpy/readers/hrit_base.py

```diff
@@ -327,13 +327,25 @@ def read_band(self, key, info):
         elif self.mda['number_of_bits_per_pixel'] in [8, 10]:
             dtype = np.uint8
         shape = (shape, )
-        # For reading the image data, unzip_context is faster than generic_open
-        with utils.unzip_context(self.filename) as fn:
-            data = np.memmap(fn, mode='r',
-                             offset=self.mda['total_header_length'],
-                             dtype=dtype,
-                             shape=shape)
-            data = da.from_array(data, chunks=shape[0])
+        # check, whether 'filename' is a str, thus path to a file on disk,
+        #  or a file like obj, possibly residing already in memory
+        if isinstance(self.filename, str):
+            # For reading the image data, unzip_context is faster than generic_open
+            with utils.unzip_context(self.filename) as fn:
+                data = np.memmap(fn, mode='r',
+                                 offset=self.mda['total_header_length'],
+                                 dtype=dtype,
+                                 shape=shape)
+        else:  # filename is likely to be a file-like object
+            with utils.generic_open(self.filename, mode="rb") as fp:
+                no_elements = np.prod(shape)
+                fp.seek(self.mda['total_header_length'])
+                data = np.frombuffer(
+                    fp.read(np.dtype(dtype).itemsize * no_elements),
+                    dtype=np.dtype(dtype),
+                    count=no_elements.item()
+                ).reshape(shape)
```

Comment:
The `isinstance` check could be replaced with a `try-catch` block; first trying to use memmap via unzip_context, and frombuffer via generic_open otherwise.